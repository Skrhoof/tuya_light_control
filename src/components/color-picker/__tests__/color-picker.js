import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ColorPicker from '../color-picker';

describe('ColorPicker Component', () => {
  it('basic render', () => {
    const component = renderer.create(
      <ColorPicker
        style={{}}
        hsb={[360, 100, 100]}
        disabled={false}
        onValueChangeComplete={jest.fn()}
      />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('component update', () => {
    const component = renderer.create(
      <ColorPicker
        style={{}}
        hsb={[360, 100, 100]}
        disabled={false}
        onValueChangeComplete={jest.fn()}
      />
    );

    component.update(
      <ColorPicker
        style={{}}
        width={400}
        height={400}
        hsb={[360, 100, 100]}
        disabled={true}
        onValueChangeComplete={jest.fn()}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

});
