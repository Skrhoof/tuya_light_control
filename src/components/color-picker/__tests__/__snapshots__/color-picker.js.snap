// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Component basic render 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 608,
        "justifyContent": "center",
        "width": 608,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 528,
        "justifyContent": "center",
        "width": 528,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 528,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 528,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 64,
          "height": 128,
          "justifyContent": "center",
          "width": 128,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F8F8F8",
            "borderColor": "#fff",
            "borderRadius": 64,
            "borderWidth": 8,
            "height": 112,
            "width": 112,
          }
        }
      />
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "borderRadius": 64,
            "height": 128,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": 128,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 40,
        "height": 80,
        "justifyContent": "center",
        "left": 528,
        "opacity": 1,
        "position": "absolute",
        "top": 264,
        "width": 80,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FF0000",
          "borderRadius": 28,
          "height": 56,
          "width": 56,
        }
      }
    />
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 80,
        }
      }
    />
  </View>
  <View
    pointerEvents="box-only"
    style={
      Object {
        "height": 608,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 608,
      }
    }
  />
</View>
`;

exports[`ColorPicker Component component update 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 480,
        "justifyContent": "center",
        "width": 480,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 400,
        "justifyContent": "center",
        "width": 400,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 400,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 400,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 64,
          "height": 128,
          "justifyContent": "center",
          "width": 128,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F8F8F8",
            "borderColor": "#fff",
            "borderRadius": 64,
            "borderWidth": 8,
            "height": 112,
            "width": 112,
          }
        }
      />
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "borderRadius": 64,
            "height": 128,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": 128,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 40,
        "height": 80,
        "justifyContent": "center",
        "left": 400,
        "opacity": 1,
        "position": "absolute",
        "top": 200,
        "width": 80,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FF0000",
          "borderRadius": 28,
          "height": 56,
          "width": 56,
        }
      }
    />
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 80,
        }
      }
    />
  </View>
  <View
    pointerEvents="box-only"
    style={
      Object {
        "height": 480,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 480,
      }
    }
  />
</View>
`;
